src/Button/index.jsx

```diff
@@ -4,40 +4,65 @@ import PropTypes from 'prop-types';
 
 import styles from './Button.scss';
 
-function Button(props) {
-  const {
-    buttonType,
-    className,
-    label,
-    inputRef,
-    isClose,
-    onBlur,
-    onClick,
-    onKeyDown,
-    type,
-    ...other
-  } = props;
+class Button extends React.Component {
+  constructor(props) {
+    super(props);
 
-  return (
-    <button
-      className={classNames([
-        ...className,
-        styles.btn,
-      ], {
-        [styles[`btn-${buttonType}`]]: buttonType !== undefined,
-      }, {
-        [styles.close]: isClose,
-      })}
-      onBlur={onBlur}
-      onClick={onClick}
-      onKeyDown={onKeyDown}
-      type={type}
-      ref={inputRef}
-      {...other}
-    >
-      {label}
-    </button>
-  );
+    const {
+      onBlur,
+      onKeyDown,
+    } = props;
+
+    this.onBlur = onBlur.bind(this);
+    this.onKeyDown = onKeyDown.bind(this);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick(e) {
+    this.buttonRef.focus();
+    this.props.onClick(e);
+  }
+
+  getRefs(input) {
+    this.props.inputRef(input);
+    this.buttonRef = input;
+  }
+
+  render() {
+    const {
+      buttonType,
+      className,
+      label,
+      inputRef,
+      isClose,
+      onBlur,
+      onClick,
+      onKeyDown,
+      type,
+      ...other
+    } = this.props;
+
+    return (
+      <button
+        className={classNames([
+          ...this.props.className,
+          styles.btn,
+        ], {
+          [styles[`btn-${this.props.buttonType}`]]: this.props.buttonType !== undefined,
+        }, {
+          [styles.close]: this.props.isClose,
+        })}
+        onBlur={this.props.onBlur}
+        onClick={this.onClick}
+        onKeyDown={this.props.onKeyDown}
+        type={this.props.type}
+        ref={(input) => { this.getRefs(input); }}
```

Comment:
Will this create a new function each render? If so, then I think this will force a re-render for every button since React's shallow equality check will fail.

[Here are some more examples](https://github.com/vasanthk/react-bits/blob/master/gotchas/01.pure-render-checks.md#bad-1).